test/tune/test_tune.py

```diff
@@ -352,6 +386,64 @@ def evaluate_config_empty(config):
     )
 
 
+def test_passing_search_alg():
+    from flaml import tune
+
+    # search_space
+    so_search_space = {
+        "steps": 100,
+        "width": tune.uniform(0, 20),
+        "height": tune.uniform(-100, 100),
+    }
+    mo_search_space = {
+        "x1": tune.uniform(lower=0.000001, upper=1.0),
+        "x2": tune.uniform(lower=0.000001, upper=1.0),
+    }
+
+    # lexicographic objectives
+    lexico_objectives = {}
+    lexico_objectives["metrics"] = ["brain", "currin"]
+    lexico_objectives["tolerances"] = {"brain": 10.0, "currin": 0.0}
+    lexico_objectives["targets"] = {"brain": 0.0, "currin": 0.0}
+    lexico_objectives["modes"] = ["min", "min"]
+
+    ## Passing search_alg through string
+    # Non lexico tune
+    tune.run(
+        _easy_objective,
+        search_alg="BlendSearch",
+        metric="mean_loss",
+        mode="min",
+        num_samples=10,
+        config=so_search_space,
+    )
+    # lexico tune
+    tune.run(
+        _BraninCurrin, search_alg="CFO", num_samples=10, config=mo_search_space, lexico_objectives=lexico_objectives
+    )
+
+    ## Passing search_alg through instance
+    so_bs = BlendSearch(time_budget_s=5, metric="mean_loss", mode="min")
+    mo_bs = CFO(time_budget_s=5)
```

Comment:
mo_bs is CFO? Add a note here if you plan to change CFO into BlendSearch in future.